packages/core/src/hooks/useNotifications.ts

```diff
@@ -1,10 +1,24 @@
 import { useMemo } from 'react'
-import { useNotificationsContext } from '../providers'
+import { Notification, useNotificationsContext } from '../providers'
 import { useEthers } from './useEthers'
+import { useInterval } from './useInterval'
 
-export function useNotifications() {
+interface NotificationsOptions {
+  checkInterval?: number
+  expirationPeriod?: number
+}
+
+function getExpiredNotifications(notifications: Notification[], expirationPeriod: number) {
+  const timeFromCreation = (creationTime: number) => Date.now() - creationTime
+
+  return notifications.filter((notification) => timeFromCreation(notification.submittedAt) >= expirationPeriod)
+}
```

Comment:
We could push implementation further into functional direction for fun and readability:
```ts
function getExpiredNotifications(notifications: Notification[], expirationPeriod: number) {
  const isExpired = (expirationTime: number) => 
    (notification: Notification) => 
      creationTime >= expirationTime
  return notifications.filter(isExpired(Time.now() - expirationPeriod))
}
```